ui/src/api/base.js

```diff
@@ -1,13 +1,37 @@
 import axios from 'axios';
+import { goTo } from '@history';
+import { AccountsRoutes } from '@routes/urls';
 
 const apiConfig = {
   baseURL:
     process.env.NODE_ENV === 'production'
       ? '/api'
-      : 'https://127.0.0.1:5000/api',
+      : `${process.env.REACT_APP_BACKEND_BASE_URL}/api`,
   withCredentials: true,
 };
 
+const URLS_NOT_TO_REDIRECT_IF_UNAUTHORIZED = ['/me'];
+
+const url_should_not_redirect = url => {
+  return URLS_NOT_TO_REDIRECT_IF_UNAUTHORIZED.some(val => url.endsWith(val));
+};
+
 const http = axios.create(apiConfig);
 
-export { http, apiConfig };
+const responseRejectInterceptor = error => {
```

Comment:
What about something like that, (I am not a big fan of double negation, it can be inverted and cleaned further), but like this I can follow the conditions easily about when a redirect will happen

```js
  if (
    error.response &&
    error.response.status === 401 &&
    !url_should_not_redirect(error.response.request.responseURL)
  ) {
    goTo(`${AccountsRoutes.login}?sessionExpired`);
  }
  return Promise.reject(error);
```